### ISSUE TITLE
[Enhancement] Support for fallbacks when loading cloudpickle items fail

### ISSUE THREAD

**Issue by Miffyli** [DLR-RM/stable-baselines3#172]
Related issue #171 

As seen in #171, cloudpickle/pickle can easily fail when transferring models between Python versions (and in case some other shenanigans). There is currently no way to address this issue as loading of model tries to un-pickle all pickle objects and does not catch errors.

Two suggestions to deal with this, both of which would be nice to have:
1) Provide `custom_objects` dictionary (similar to what we had in [SB2](https://stable-baselines.readthedocs.io/en/master/modules/a2c.html#stable_baselines.a2c.A2C.load)), which will override any objects read from file. This could be used to even prevent loading the item from the file, in which case all JSON-ed items can be used as before.
2) Update `set_parameters` function to only load parameters from the file but not the `data` objects (which include pickled items). This was included in SB2.

@ManifoldFR If you had other suggestions please feel free to throw them here :)

**Comment by ManifoldFR** [comment]
From what I saw the problem I encountered is with the learning rate and clip ratio schedulers (for PPO), in the particular case where a constant scheduler is crated using `constant_fn` (I don't know about other cases though) because pickling/unpickling functions is brittle. I think we can look at how PyTorch supports saving state dicts for [schedulers](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py). From the source I see they use classes and the `__dict__` attribute to save/load state dicts.

**Comment by ManifoldFR** [comment]
I started a branch for schedulers. I think we should *not* subclass from PyTorch's `_LRScheduler` straightaway to get the learning rate because they're designed to step with the optimizer.
I think we can take inspiration from their design though:
* use a class with `state_dict`/`load_state_dict` methods
* update the scheduler parameters using their `step()` method, update the param groups from there

**Comment by Miffyli** [comment]
I wouldn't just focus on schedules though, as other objects are stored as pickled objects (no easy alternative). I am not sure having non-pickled versions of all objects is possible, and it would not be easy at the very least. I'd focus on the fallback tools to help reconstructing models if loading fails (part of the reason why currently the saved model contains some description of the pickled object). However if you can find something that could be used to achieve less-brittle serialization I am all ears!

**Comment by aliamiri1380** [comment]
this can create a file from model, but it's too large

```
import dill
with open('nnn.pkl', 'wb') as f:
    dill.dump(model, f)
```

**Comment by araffin** [comment]
as a follow-up, I added support for custom objects here: https://github.com/DLR-RM/stable-baselines3/pull/336
and included that in the rl zoo: https://github.com/DLR-RM/rl-baselines3-zoo/pull/69
(so models trained with python 3.6/3.7 can be loaded with python 3.8+, but not retrained yet)

**Comment by MaximilienLC** [comment]
> as a follow-up, I added support for custom objects here: #336 and included that in the rl zoo: [DLR-RM/rl-baselines3-zoo#69](https://github.com/DLR-RM/rl-baselines3-zoo/pull/69) (so models trained with python 3.6/3.7 can be loaded with python 3.8+, but not retrained yet)

Hi, I might have missed it but how is it then possible to escape the Assertion Error and load the models on python 3.8?
